tests/test_dynamics_bookmarks.py

```diff
@@ -0,0 +1,213 @@
+
+        return_value["start_date"] = self.start_date
+
+        return return_value
+
+    @staticmethod
+    def convert_state_to_utc(date_str):
+        """
+        Convert a saved bookmark value of the form '2020-08-25T13:17:36-07:00' to
+        a string formatted utc datetime,
+        in order to compare against json formatted datetime values
+        """
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return datetime.datetime.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
+    def calculated_states_by_stream(self, current_state):
+        """
+        Look at the bookmarks from a previous sync and set a new bookmark
+        value based off timedelta expectations. This ensures the subsequent sync will replicate
+        at least 1 record but, fewer records than the previous sync.
+
+        Sufficient test data is required for this test to cover a given stream.
+        An incremental replication stream must have at least two records with
+        replication keys that differ by more than the lookback window.
+
+        If the test data is changed in the future this will break expectations for this test.
+
+        The following streams barely make the cut:
+        account         "2021-02-09T18:17:30.000000Z"
+                        "2021-02-09T16:24:58.000000Z"
+        contact         "2021-02-09T18:17:41.000000Z"
+                        "2021-02-09T17:10:09.000000Z"
+        """
+        timedelta_by_stream = {stream: [1,0,0]  # {stream_name: [days, hours, minutes], ...}
+                               for stream in self.expected_streams()}
+
+        stream_to_calculated_state = {stream: "" for stream in current_state['bookmarks'].keys()}
+        for stream, state in current_state['bookmarks'].items():
+            state_key, state_value = next(iter(state.keys())), next(iter(state.values()))
+            state_as_datetime = dateutil.parser.parse(state_value)
+
+            days, hours, minutes = timedelta_by_stream[stream]
+            calculated_state_as_datetime = state_as_datetime - datetime.timedelta(days=days, hours=hours, minutes=minutes)
+
+            state_format = '%Y-%m-%dT%H:%M:%S-00:00'
+            calculated_state_formatted = datetime.datetime.strftime(calculated_state_as_datetime, state_format)
+
+            stream_to_calculated_state[stream] = {state_key: calculated_state_formatted}
+
+        return stream_to_calculated_state
+
+
+    def test_run(self):
+        expected_streams =  self.expected_streams()
+
+        expected_replication_keys = self.expected_replication_keys()
+        expected_replication_methods = self.expected_replication_method()
+
+        ##########################################################################
+        ### First Sync
+        ##########################################################################
+
+        conn_id = connections.ensure_connection(self, original_properties=False)
```

Comment:
We are use the False flag but not overwriting the start_date. When you pass False here it will pass False to `get_properties` as well. It should only be used when we want to use a non-default value in our report card. One that would not apply to the other tests.